mozilla_version/gecko.py

```diff
@@ -333,6 +349,71 @@ def _compare(self, other):
     def _compare_version_type(self, other):
         return self.version_type.compare(other.version_type)
 
+    def bump(self, field):
+        """Bump the number defined `field`.
+
+        This function handles previous edge cases.
+
+        Returns:
+            A new GeckoVersion with the right field bumped and the following ones set to 0,
+            if they exist.
+
+        """
+        if field == 'build_number' and self.build_number is None:
+            raise ValueError('Cannot bump the build number if it is not already set')
+        elif field == 'major_number' and self.is_esr:
+            raise ValueError('Cannot predictably know the next major ESR number')
+        elif (
+            (
+                field in (
+                    'minor_number', 'patch_number', 'release_candidate_number', 'old_fourth_number'
+                ) and
+                any((self.is_nightly, self.is_aurora_or_devedition, self.is_beta))
+            ) or
+            field == 'beta_number' and any((self.is_nightly, self.is_aurora_or_devedition))
+        ):
+            raise ValueError('Version type "{}" cannot bump "{}"'.format(self.version_type, field))
```

Comment:
Wait, I'm confused, why bumping `minor_number` with `is_nightly` isn't an option? AFAIK, esr version right now is `68.1a1` and bumping that means `68.2a1` <= this is bumping the minor number and also is nightly 🤔 I may be missing something I believe

Reply:
Ooooops. I had thought about this case, but I forgot it. Thanks for pointing it out!